tests/Annotation/EntityTest.php

```diff
@@ -0,0 +1,120 @@
+    {
+        $reader = new AnnotationReader();
+        return $reader->getClassAnnotation(
+            new \ReflectionClass(AllAvailableActionsAnnotatedModel::class),
+            Entity::class
+        );
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function isAnnotation()
+    {
+        $this->assertIsAnnotation(Entity::class);
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function entityReturnProperActionForGivenOptions()
+    {
+        $actions = ['addable', 'getable', 'listable', 'editable', 'deletable'];
+        $reader = new AnnotationReader();
+        $entity = new Entity(['actions' => $actions]);
+        $this->assertEquals($actions, $entity->getActions());
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function entityReturnAllCorrectActionForGivenOptions()
+    {
+        $actions = ['addable', 'getable', 'listable', 'editable', 'deletable', 'adorable'];
+        $reader = new AnnotationReader();
+        $entity = new Entity(['actions' => $actions]);
+        $this->assertNotEquals($actions, $entity->getActions());
+        $properActions = ['addable', 'getable', 'listable', 'editable', 'deletable'];
+        $this->assertEquals($properActions, $entity->getActions());
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     **/
+    public function entityAnnotationIsUsed()
+    {
+        $metadata = $this->getMetadata();
+        $this->assertNotNull($metadata);
+        $this->assertNotNull($metadata->getActions());
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function entityHasProperActions()
+    {
+        $actions = ['addable', 'deletable', 'editable', 'getable', 'listable'];
+        $metadata = $this->getMetadata();
+        $this->assertEquals($actions, $metadata->getActions());
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function annotationGetAnnotationAvailableActions()
+    {
+        $actions = ['addable', 'getable', 'listable', 'editable', 'deletable'];
+        $reader = new AnnotationReader();
+        $entity = new Entity([]);
+        $this->assertEquals($actions, ObjectManipulator::callProtectedMethod(
+            $entity,
+            'getAvailableActions'
+        ));
+    }
+
+    /**
+     * @group EntityAnnotation
+     * @group Annotation
+     * @test
+     */
+    public function enableActionWorkForGivenAddAvailableAction()
+    {
+        $entity = new Entity([]);
+        ObjectManipulator::callProtectedMethod($entity, 'addAvailableAction', ['action' => 'adorable']);
+        ObjectManipulator::callProtectedMethod($entity, 'enableAction', ['action' => 'adorable']);
+        $this->assertEquals(['adorable'], $entity->getActions());
```

Comment:
are you testing protected method? :) don't

Reply:
fixed